docs/world api.md

```diff
@@ -339,6 +314,60 @@ avoiding the need for indirect conditions at the expense of performance.
 An item rule is a function that returns `True` or `False` for a `Location` based on a single item. It can be used to
 reject the placement of an item there.
 
+### Events (or "generation-only items/locations")
+
+An event item or location is one that only exists during multiworld generation. The server is never made aware of them,
+event locations can never be checked, nor can event items be received during play. Events are used to represent in-game
+actions (that aren't regular Archipelago locations) when either:
+
+* we want to show in the spoiler log when the player is expected to perform the in-game action, or
+* it's the cleanest way to represent how that in-game action impacts logic
+
+Typical examples include completing the goal, defeating a boss, or flipping a switch that changes several areas.
+
+To be precise: the term "event" on its own refers to the special combination of an "event item" placed on an "event
+location". Event items and locations are created the same way as normal items and locations, except that they have an
+`id` of `None`, they must not exist in the `name_to_id` lookups, and an event item must be placed on an event location
+(and vice versa). By convention, the event location and item typically have the same name, though this is not a
+requirement. Finally, although events are often described as "fake" items and locations, it's important to understand
```

Comment:
```suggestion
(and vice versa). Finally, although events are often described as "fake" items and locations, it's important to understand
```
I don't think this is a good convention -- since it gets shown in the spoiler log, there should be an attempt to make it more readable than just "Victory: Victory"

Reply:
Honestly, I agree. But I felt trying to change recommendations/policy would be out of scope in a PR about making these docs easier to understand.

Reply:
I'm one of the docs maintainers so you have my approval to make this change o:

Reply:
All right then. I've pushed another commit with just the "different names" changes.

Reply:
This must be some kind of record; it still hasn't processed the commit

![image](https://github.com/user-attachments/assets/932df79a-0364-4e29-90a8-92fd8b468c07)